addon/-private/system/model/internal-model.js

```diff
@@ -584,6 +644,15 @@ InternalModel.prototype = {
     });
   },
 
+  _resetRecord() {
```

Comment:
Can you put a comment saying that this doesn't reset relationships/attributes on purpose, given that constraint, the name is a bit unfortunate